### PR TITLE
Add a container display to dispenser UI

### PR DESCRIPTION
**Pull request by Tayrtahn**
## About the PR

Added a sprite viewer to the reagent dispenser UI window that shows the current appearance of the container inserted into the machine.

## Why / Balance

It's fun and more intuitive to see the container fill up than just reading the numbers going up on the UI. This is especially fun for mixing drinks in metamorphic glasses, since you get visual feedback when making the right mix.

## Technical details

Adds a SpriteView control to ReagentDispenserWindow. ReagentDispenserSystem is modified to pass the NetEntity ID of the output container from the server to the client UI.

This is the first time I've messed with UI stuff, so let me know what's wrong so I can fix it.

I cleaned up a few compiler warnings in the files I touched too, mostly formatting issues.

## Media

Chem dispenser:

https://github.com/space-wizards/space-station-14/assets/85356/6e13bdd2-1522-4529-b341-6f79d84ceb95

Booze/Soda dispenser (in combination with #25319):

https://github.com/space-wizards/space-station-14/assets/85356/b601cf0d-a604-40c6-93ba-3bd402480237

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Added a live preview display to reagent dispensers.

### CODE REVIEW

**Comment by UbaserB** [comment]
wow this is really cool

**Comment by deltanedas** [comment]
aluminium sulfuric acid and welding fuel lived in harmony

until the space lube nation attacked

File: Content.Client/Chemistry/UI/ReagentDispenserWindow.xaml.cs [hunk 1]
```diff
@@ -86,6 +85,11 @@ public void UpdateState(BoundUserInterfaceState state)
             UpdateContainerInfo(castState);
             UpdateReagentsList(castState.Inventory);
 
+            if (castState.OutputContainerEntity is { } containerEntity)
+                View.SetEntity(containerEntity);
+            else
+                View.SetEntity(null);
```
Comment (EmoGarbage404) [inline Content.Client/Chemistry/UI/ReagentDispenserWindow.xaml.cs:91]:
```suggestion
            View.SetEntity(castState.OutputContainerEntity);
```
nullable args moment

**Review by EmoGarbage404 — APPROVED** [review]

File: Content.Client/Chemistry/UI/ReagentDispenserWindow.xaml.cs (hunk 1 above)
Comment (Tayrtahn) [reply Content.Client/Chemistry/UI/ReagentDispenserWindow.xaml.cs]: Weird. There's SetEntity(EntityUid? uid) and SetEntity(NetEntity netEnt). But not SetEntity(NetEntity? netEnt).

Am I gonna make an engine PR for a single question mark? Possibly.

Comment (Tayrtahn) [reply Content.Client/Chemistry/UI/ReagentDispenserWindow.xaml.cs]: Nope, I'm not, since that makes a bunch of other SetEntity calls ambiguous.

Comment (Tayrtahn) [reply Content.Client/Chemistry/UI/ReagentDispenserWindow.xaml.cs]: So we'll TryGetEntity the netent first, then use that nullable as the argument to SetEntity. Means we need to add entity manager as a dependency, but it works!

**Pull request merged**